client/src/features/ProjectPageV2/ProjectPageContent/Settings/ProjectKeywordsFormField.tsx

```diff
@@ -0,0 +1,122 @@
+import cx from "classnames";
+import { Controller } from "react-hook-form";
+import { Button, FormText, Label } from "reactstrap";
+import KeywordContainer from "~/components/keywords/KeywordContainer";
+import KeywordBadge from "~/components/keywords/KeywordBadge";
+import { PlusLg } from "react-bootstrap-icons";
+import type {
+  FieldErrors,
+  UseFormGetValues,
+  UseFormSetValue,
+  Control,
+} from "react-hook-form";
+import type { ProjectV2MetadataWithKeyword } from "../../settings/projectSettings.types";
+
+interface ProjectKeywordsFormFieldProps {
+  control: Control<Required<ProjectV2MetadataWithKeyword>>;
+  errors: FieldErrors<Required<ProjectV2MetadataWithKeyword>>;
+  getValues: UseFormGetValues<Required<ProjectV2MetadataWithKeyword>>;
+  oldKeywords?: string[];
+  setValue: UseFormSetValue<Required<ProjectV2MetadataWithKeyword>>;
+}
+
+export default function ProjectKeywordsFormField({
+  control,
+  errors,
+  getValues,
+  oldKeywords,
+  setValue,
+}: ProjectKeywordsFormFieldProps) {
+  return (
+    <div>
+      <Label className="form-label" for="project-keyword">
+        Keywords
+      </Label>
+      <div className={cx("input-group", "input-group-sm", "mb-2")}>
+        <Controller
+          control={control}
+          name="keyword"
+          render={({ field }) => (
+            <>
+              <input
+                id="project-keyword"
+                placeholder="Add new keyword"
+                type="string"
+                {...field}
+                className={cx("form-control", errors.keyword && "is-invalid")}
```

Comment:
The `error` state is found in `fieldState`:
```suggestion
                className={cx("form-control", fieldState.error && "is-invalid")}
```

Also, it doesn't seem that this input is validated, so maybe we can remove error styling here.